java/test/com/google/jglome/TestVector.java

```diff
@@ -0,0 +1,96 @@
+package com.google.jglome;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.math.BigInteger;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * A helper class to generate input data from GLOME test vectors.
+ */
+public class TestVector {
+
+  private byte[] ka; // A's public key
+  private byte[] kb; // B's public key
+  private byte[] kah; // A's ephemeral private key
+  private byte[] kbh; // B's private key
+  private byte[] ks; // shared secret
+  private byte[] tag; // generated tag
+  private byte[] msg; // message to be sent
+  private int cnt; // counter of messaged that has been sent from the sending side to the receiving
+
+  public byte[] getKa() {
+    return ka;
+  }
+
+  public byte[] getKb() {
+    return kb;
+  }
+
+  public byte[] getKah() {
+    return kah;
+  }
+
+  public byte[] getKbh() {
+    return kbh;
+  }
+
+  public byte[] getKs() {
+    return ks;
+  }
+
+  public byte[] getTag() {
+    return tag;
+  }
+
+  public byte[] getMsg() {
+    return msg;
+  }
+
+  public int getCnt() {
+    return cnt;
+  }
+
+  TestVector(String ka, String kb, String kah, String kbh, String tag, String ks, String msg,
+      int cnt) {
+    this.ka = fromHexString(ka);
+    this.kb = fromHexString(kb);
+    this.kah = fromHexString(kah);
+    this.kbh = fromHexString(kbh);
+    this.ks = fromHexString(ks);
+    this.tag = fromHexString(tag);
+    this.msg = msg.getBytes(UTF_8);
+    this.cnt = cnt;
+  }
+
+  private static byte[] fromHexString(String str) {
+    byte[] bytes = new BigInteger("10" + str.replaceAll("\\s", ""), 16).toByteArray();
+    return Arrays.copyOfRange(bytes, 1, bytes.length);
+  }
```

Comment:
That's a cool hack, but it took me a bit to figure it out - could you please add a comment that explains why this works (in particular the "add prefix/strip prefix" aspect)?

Reply:
Fixed